modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1093,7 +1093,16 @@ void GDScriptAnalyzer::resolve_class_member(GDScriptParser::ClassNode *p_class,
 							}
 						}
 						if (is_get_node) {
-							parser->push_warning(member.variable, GDScriptWarning::GET_NODE_DEFAULT_WITHOUT_ONREADY, is_using_shorthand ? "$" : "get_node()");
+							GDScriptParser::GetNodeNode *get_node_node = static_cast<GDScriptParser::GetNodeNode *>(expr);
+							String offending_syntax = "get_node()";
+							if (is_using_shorthand) {
+								if (get_node_node->full_path.begins_with("%")) {
```

Comment:
There is `use_dollar` property:

https://github.com/godotengine/godot/blob/eee39f004b42a4948af90cdebedf65c34a4a4970/modules/gdscript/gdscript_parser.h#L877

Compare: `%Node` and `$"%Node"`. Also please add a test.

Reply:
Thank you for the `use_dollar` suggestion. I'm using it in a newer version of the code that I'll push very shortly.

As for the test, I will try to get one added soon!